### PR TITLE
Add Login page

### PR DESCRIPTION
**Pull request by raidnav**
- wrap gorm.db to controller
- add login

### CODE REVIEW

File: repositories/User.go [hunk 1]
```diff
@@ -26,9 +26,13 @@ func (u *User) BeforeSave() error {
 	return nil
 }
 
-func (u *User) FindUser(db *gorm.DB, uid int32) (*User, error) {
+func (u *User) FindUser(db *gorm.DB, args map[string]string) (*User, error) {
 	var err error
-	err = db.Debug().Model(User{}).Where("id = ?", uid).Take(&u).Error
+	var queryBuilder = ""
+	for field, value := range args {
+		queryBuilder += field + " = " + value
```
Comment (nurlailifajriyah) [inline repositories/User.go]: check this again

File: repositories/User.go [hunk 2]
```diff
@@ -26,9 +26,18 @@ func (u *User) BeforeSave() error {
...
+func (u *User) FindUser(db *gorm.DB, args map[string]string) (*User, error) {
 	var err error
-	err = db.Debug().Model(User{}).Where("id = ?", uid).Take(&u).Error
+	var queryBuilder = ""
+	var keys = len(args) - 1
+	var iter = 0
+	for field, value := range args {
+		queryBuilder += field + " = " + value
+		if iter < keys {
+			queryBuilder += " AND "
```
Comment (nurlailifajriyah) [inline repositories/User.go]: check the query builder one more time

File: repositories/User.go [hunk 3]
```diff
@@ -26,9 +26,19 @@ func (u *User) BeforeSave() error {
...
+	var queryBuilder = ""
+	var keys = len(args) - 1
+	var iter = 0
+	for field, value := range args {
+		queryBuilder += field + " = '" + value + "'"
+		if iter < keys {
+			queryBuilder += " AND "
+			iter++
+		}
+	}
```
Comment (raidnav) [inline repositories/User.go]: Actually, this is a tech-debt. 😂

Comment (nurlailifajriyah) [reply repositories/User.go]: I know

Comment (nurlailifajriyah) [reply repositories/User.go]: change it later!!

**Pull request merged**